spark/src/test/scala/org/apache/comet/CometFuzzTestSuite.scala

```diff
@@ -188,6 +188,22 @@ class CometFuzzTestSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("regexp_replace") {
+    withSQLConf(CometConf.COMET_REGEXP_ALLOW_INCOMPATIBLE.key -> "true") {
+      val df = spark.read.parquet(filename)
+      df.createOrReplaceTempView("t1")
+      // We want to make sure that the schema generator wasn't modified to accidentally omit
+      // StringType, since then this test would not run any queries and silently pass.
+      var testedString = false
+      for (field <- df.schema.fields if field.dataType == StringType) {
+        testedString = true
+        val sql = s"SELECT regexp_replace(${field.name}, 'a', 'b') FROM t1"
```

Comment:
I'm not sure how likely it is that the input strings will contain the character `a`, so it is possible that this test isn't really testing anything. I modified the test locally to add:

```
        val sql = s"SELECT count(*) FROM t1 WHERE ${field.name} LIKE '%a%'"
        spark.sql(sql).show()
```

this produced

```
+--------+
|count(1)|
+--------+
|       0|
+--------+
```

This is a general problem with just comparing results between Spark and Comet and it would be nice to have a way to also ensure that we see expected results.

Reply:
Perhaps we should improve the logic in `ParquetGenerator` to also generate ASCII strings:

```scala
      case DataTypes.StringType =>
        Range(0, numRows).map(_ => {
          r.nextInt(10) match {
            case 0 if options.allowNull => null
            case 1 => r.nextInt().toByte.toString
            case 2 => r.nextLong().toString
            case 3 => r.nextDouble().toString
            case _ => r.nextString(8)
          }
        })
```

The call to ` r.nextString(8)` generates strings containing characters in the range 0 through 0xD800, so while it could theoretically generate `a`, it will be rare.

Reply:
Makes sense. I find myself expanding the generator for #1697 to create dictionaries, so let me sync up with you for good solutions to expand the schemas and values in specific types.

Reply:
Pushed a commit to generate ASCII strings, ran your `COUNT(*)` query and confirmed I saw at least a dozen matching rows.